### PR TITLE
feat: Add mobile navbar

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Related to: #3185

Screen Shot 2022-07-25 at 13 41 06
Screen Shot 2022-07-25 at 13 41 29
Screen Shot 2022-07-25 at 13 41 18

### CODE REVIEW

File: site/src/components/NavbarView/NavbarView.tsx [hunk 1]
```diff
@@ -19,41 +23,66 @@ export const Language = {
   users: "Users",
 }
 
-export const NavbarView: React.FC<NavbarViewProps> = ({ user, onSignOut }) => {
+const NavItems: React.FC<{ className?: string; linkClassName?: string }> = ({ className }) => {
```
Comment (Kira-Pilot) [inline site/src/components/NavbarView/NavbarView.tsx:26]: You can just import `FC` as a standalone if you wish

File: site/src/components/NavbarView/NavbarView.tsx [hunk 2]
```diff
@@ -19,41 +23,66 @@ export const Language = {
...
   const styles = useStyles()
   const location = useLocation()
+
+  return (
+    <List className={combineClasses([styles.navItems, className])}>
```
Comment (Kira-Pilot) [inline site/src/components/NavbarView/NavbarView.tsx:31]: What's the benefit of using `combineClasses` instead of template literals?

**Comment by Kira-Pilot** [comment]
I think we can add a storybook for this at a smaller screensize: https://storybook.js.org/docs/react/essentials/viewport

**Review by Kira-Pilot — APPROVED** [review]

File: site/src/components/NavbarView/NavbarView.tsx (hunk 2 above)
Comment (BrunoQuaresma) [reply site/src/components/NavbarView/NavbarView.tsx:31]: It handles, `null`, `undefined` and conditional styles like `{ isActive: styles.active }`.

File: site/src/components/NavbarView/NavbarView.tsx [hunk 3]
```diff
@@ -19,41 +23,66 @@ export const Language = {
...
+          {Language.users}
+        </NavLink>
+      </ListItem>
+    </List>
+  )
+}
+
+export const NavbarView: React.FC<NavbarViewProps> = ({ user, onSignOut }) => {
+  const styles = useStyles()
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false)
```
Comment (AbhineetJain) [inline site/src/components/NavbarView/NavbarView.tsx:56]: `nit` Maybe worth a discussion in FE Variety, but I think by convention the setter function need not have the verb like `is` or `has`. For example, I would use `setDrawerOpen` here.

File: site/src/components/NavbarView/NavbarView.tsx [hunk 4]
```diff
@@ -124,13 +173,25 @@ const useStyles = makeStyles((theme) => ({
...
+        height: "100%",
         background: theme.palette.secondary.dark,
-        right: theme.spacing(3),
-        height: 2,
         position: "absolute",
+
+        [theme.breakpoints.up("md")]: {
+          bottom: 0,
+          left: theme.spacing(3),
+          width: `calc(100% - 2 * ${theme.spacing(3)}px)`,
```
Comment (AbhineetJain) [inline site/src/components/NavbarView/NavbarView.tsx:185]: `nit` A comment describing the calculation here would be nice.

**Review by AbhineetJain — APPROVED** [review]
Nice start to responsive design! This is exciting. 🙌

File: site/src/components/NavbarView/NavbarView.tsx (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/components/NavbarView/NavbarView.tsx:56]: I think this would be nice!

File: site/src/components/NavbarView/NavbarView.tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply site/src/components/NavbarView/NavbarView.tsx:185]: I put the commit on auto merge but I will address this when I touch it again.

**Pull request merged**